core/ledger/kvledger/txmgmt/txmgr/lockbasedtxmgr/collection_val_ext_test.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package lockbasedtxmgr
+
+import (
+	"testing"
+
+	"github.com/hyperledger/fabric/core/ledger"
+	"github.com/hyperledger/fabric/core/ledger/kvledger/txmgmt/version"
+	"github.com/hyperledger/fabric/core/ledger/mock"
+	"github.com/hyperledger/fabric/protos/common"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTransientCollectionValidation(t *testing.T) {
+	testEnv := testEnvsMap[levelDBtestEnvName]
+	testEnv.init(t, "testLedger", nil)
+	defer testEnv.cleanup()
+	txMgr := testEnv.getTxMgr()
+
+	populateTransientCollConfigForTest(t, txMgr.(*LockBasedTxMgr),
+		[]collConfigkey{
+			{"ns1", "coll1"},
+		},
+		version.NewHeight(1, 1),
+	)
+
+	sim, err := txMgr.NewTxSimulator("tx-id1")
+	assert.NoError(t, err)
+
+	const key1 = "key1"
+	const key2 = "key1"
+
+	_, err = sim.GetPrivateData("ns1", "coll1", key1)
+	assert.NoError(t, err)
+
+	_, err = sim.GetPrivateDataMultipleKeys("ns1", "coll1", []string{key1, key2})
+	assert.NoError(t, err)
+}
+
+func populateTransientCollConfigForTest(t *testing.T, txMgr *LockBasedTxMgr, nsColls []collConfigkey, ht *version.Height) {
```

Comment:
Rename to populateUnknownCollConfigForTest.

Reply:
done